shared/src/ledger/eth_bridge/storage/vote_tracked.rs

```diff
@@ -4,29 +4,23 @@ use crate::types::hash::Hash;
 use crate::types::storage::Key;
 
 #[allow(missing_docs)]
-pub const PREFIX_KEY_SEGMENT: &str = "eth_msgs";
-
-/// Get the key prefix corresponding to where details of seen [`EthereumEvent`]s
-/// are stored
-pub fn prefix() -> Key {
-    super::prefix()
-        .push(&PREFIX_KEY_SEGMENT.to_owned())
-        .expect("should always be able to construct this key")
-}
+pub const ETH_MSGS_PREFIX_KEY_SEGMENT: &str = "eth_msgs";
 
 const BODY_KEY_SEGMENT: &str = "body";
 const SEEN_KEY_SEGMENT: &str = "seen";
 const SEEN_BY_KEY_SEGMENT: &str = "seen_by";
 const VOTING_POWER_KEY_SEGMENT: &str = "voting_power";
 
-/// Generator for the keys under which details of an [`EthereumEvent`] is stored
-pub struct Keys {
-    /// The prefix under which the details are an [`EthereumEvent`] is stored
+/// Generator for the keys under which details of a vote tracked event is stored
+pub struct Keys<T: 'static> {
```

Comment:
Is there a plan to use this also for val set updates? I'm curious about the generic paramter.

Reply:
that would be the idea yeah

Reply:
Yeah I've put an idea in the PR description but it needs to be properly specced out still. Now thinking about it further, I don't think we absolutely need vote tallying or even including validator set update transactions in digests for the transfers of power. As some off chain relayer could just watch for protocol txs that contain the validator set signatures, and then do a relay once it had gathered enough. Without any vote extension digest ever needing to be constructed.

If we did go with tallying votes for the validator set updates, then once a validator set update had a quorum of voting power behind it, one of the nodes (probably whoever gets the reward) could "act on" the validator set update by doing the relay, and other nodes don't need to do anything else extra. 

We would need to track of at which block heights the validator set update signatures (i.e. the fake "vote extensions") were included, so the relaying node could know in which blocks to look. This could be done by changing the `seen_by` field to be a `BTreeSet<(Address, BlockHeight)>` rather than just a `BTreeSet<Address>`. i.e. we could record the heights at which validators submitted their "vote" for a validator set update, rather than just the addresses of validators that had voted for a validator set update.